packages/database/src/migrations/20210812034759-AdultMosquitoOccurrenceByFieldStation-modifies-data.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+
+import { insertObject, generateId, codeToId } from '../utilities';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+const SPECIES = ['Aedes', 'Anopheles'];
+
+const getReportCode = species => `PMOS_Distribution_Of_${species}`;
+const COUNTRY_CODES = ['FJ', 'KI', 'NR', 'NU', 'PG', 'SB', 'TV', 'TO', 'VU'];
+const PERMISSION_GROUP = 'PacMOSSI';
+const PROJECT_CODE = 'pacmossi';
+const MAP_OVERLAY_GROUP_CODE = 'Mosquito_occurrence_by_genus';
+
+const getReport = species => {
+  const popupItem = `'Total ${species} Mosquitoes found:'`;
+  return {
+    id: generateId(),
+    code: getReportCode(species),
+    config: {
+      fetch: {
+        aggregations: [
+          {
+            type: 'SUM_PER_ORG_GROUP',
+            config: {
+              dataSourceEntityType: 'field_station',
+            },
+          },
+        ],
+        dataElements: [`PMOS_${species}`],
+      },
+      transform: [
+        {
+          transform: 'select',
+          [popupItem]: '$row.value',
+          "'value'": "($row.value >0) ? 'Detected': 'Not Detected'", // null > 0 = false, interesting.
+          "'organisationUnitCode'": '$row.organisationUnit',
+        },
+      ],
+    },
+  };
+};
+
+const getMapOverlay = species => ({
+  id: getReportCode(species),
+  name: `Distribution of ${species} (field station)`,
+  userGroup: PERMISSION_GROUP,
+  dataElementCode: 'value',
+  isDataRegional: true,
+  measureBuilder: 'useReportServer',
+  measureBuilderConfig: {
+    dataSourceType: 'custom',
+    reportCode: getReportCode(species),
+  },
+  presentationOptions: {
+    values: [
+      {
+        icon: 'circle',
+        name: 'Detected',
+        value: 'Detected',
+        color: 'Red',
+        hideFromLegend: false,
+        hideFromPopup: true,
+      },
+      {
+        icon: 'circle',
+        name: 'Not Detected',
+        value: 'Not Detected',
+        color: 'Blue',
+        hideFromLegend: false,
+        hideFromPopup: true,
+      },
+      {
+        icon: 'circle',
+        name: 'No data',
+        color: 'Grey',
+        value: 'null',
+        hideFromLegend: false,
+        hideFromPopup: true,
+      },
+    ],
+    displayType: 'icon',
+    measureLevel: 'FieldStation',
+    hideFromPopup: true,
+    measureConfig: {
+      $all: {
+        type: 'icon',
+        values: [
+          {
+            icon: 'circle',
+            value: 'other',
+            hideFromLegend: true,
+          },
+          {
+            icon: 'circle',
+            name: 'No data',
+            color: 'Grey',
+            value: 'null',
+            hideFromPopup: true,
+            hideFromLegend: true,
+          },
+        ],
+        measureLevel: 'FieldStation',
```

Comment:
```suggestion
      [`'Total ${species} Mosquitoes found:'`]: {
        type: 'popup-only',
        values: [
          {
            value: 'null',
            hideFromPopup: true,
          },
        ],
        measureLevel: 'FieldStation',
```

Not 100% sure here, but could this be a case for the `popup-only` type?

Reply:
@rohan-bes Yeah I think you are right, I totally forgot `popup-only`. Thanks man.